### ISSUE TITLE
fix(cache): cacheable nodes that mutate a passed-in model/optimizer in place skip the mutation on a hit

### ISSUE THREAD

**Issue by treeleaves30760** [treeleaves30760/CodefyUI#254]
> 中文摘要：`ModelLoader`、`CheckpointLoader`、`EvaluateModel`、`Inference`、`LRScheduler` 都是可快取的，而且都會就地修改上游傳進來的 model／optimizer（`load_state_dict` 是就地寫入，`model.eval()` 是永久狀態切換，`EvaluateModel` 還會寫一筆 metric）。快取命中時這些修改全部不會發生，但節點會回報成功。麻煩的地方在於 `ModelLoader` 和 `CheckpointLoader` 的「可快取」正是 #144 特意加回去的：#144 的指紋正確處理了「讀」的過時問題，卻對「寫」隻字未提。所以這不是把 `cacheable` 改成 `False` 就好——那會直接推翻 #144。需要各自的設計。

Found while fixing #223 (a node-tree-wide audit for the #143 bug class). This is a distinct class from #143/#223: the side effect is not a file write but an **in-place mutation of an object the node was handed**.

## The gap

A cache hit returns the recorded outputs without constructing or calling the node (`backend/app/core/graph_engine.py:2045-2075`). A node that returns `{"model": model}` after mutating `model` in place looks fully described by that return value and is not: the mutation is what mattered, and on a hit it does not happen.

| Node | File | Mutation |
|---|---|---|
| `ModelLoaderNode` | `backend/app/nodes/io/model_loader_node.py:134` | `model.load_state_dict(state_dict, strict=strict)` — in-place |
| `CheckpointLoaderNode` | `backend/app/nodes/io/checkpoint_node.py:200,205,207,242,263` | `model.load_state_dict(...)`, `param_group["params"] = ...`, `optimizer.load_state_dict(...)`, `param_group["initial_lr"] = ...`, `lr_scheduler.load_state_dict(...)` |
| `EvaluateModelNode` | `backend/app/nodes/training/evaluate_model_node.py:154-155,204` | `model.to(device)`, `model.eval()` (a permanent flip), plus `context.log_metric("eval_accuracy", ...)` — a DB write that vanishes on a hit, so the chart is empty on a run reported as successful |
| `InferenceNode` | `backend/app/nodes/io/inference_node.py:51,54` | `to_device(model, device)`, `model.eval()` |
| `LRSchedulerNode` | `backend/app/nodes/training/lr_scheduler_node.py:93-110` | every torch scheduler constructor writes `initial_lr` into the passed-in `optimizer.param_groups`, and `OneCycleLR` rewrites `lr` |

Also worth a look, lower confidence: `EnvWrapperNode` (`backend/app/nodes/rl/env_wrapper_node.py:34-35`) caches a **live, mutable** `gym` env object, so a later cache hit hands back an env that a previous run has already stepped.

## Why this one is awkward

`ModelLoaderNode` and `CheckpointLoaderNode` are `cacheable = True` **because of #144**, which deliberately restored caching for file-reading nodes by folding a content fingerprint into the key. Both carry an explicit comment saying so:

```python
# #144: cacheable again -- cache_fingerprint() below folds the resolved
# ... Saving a newer checkpoint to the same path between runs is the whole
# point of resumable training.
cacheable = True
```

The fingerprint is correct and does its job: it covers the node's **read**. It says nothing about the node's **write**, and `cacheable = False` would undo #144 outright for the two nodes #144 most cared about. So the one-line answer that worked for #143 and #223 is not available here.

## Shapes worth weighing

1. **Don't mutate.** Have these nodes `copy.deepcopy` the input model/optimizer and return the copy, so the return value really is the whole effect and the cache is honest. Costs memory on large models and changes object identity, which some downstream node may rely on.
2. **`cacheable = False` for the ones where the mutation is the point** (`CheckpointLoader`, `ModelLoader`), accepting the #144 regression for them and keeping the fingerprint for the rest.
3. **Separate "reads external state" from "writes to its inputs" in the engine**, e.g. a node can be cache-*keyed* but still forced to execute. That is the general mechanism #143 option 2 gestured at and nobody has built.
4. **Per-node judgement.** `EvaluateModel`'s `log_metric` and `Inference`'s `model.eval()` may deserve different answers from `CheckpointLoader`'s `load_state_dict`.

## Also worth checking while here

`backend/tests/test_cache_writer_nodes.py:36` hardcodes `WRITER_NODES = [ImageWriterNode, ModelSaverNode, CheckpointSaverNode]`. A registry-wide invariant test — "any node whose `execute` reaches `core.checkpoints`, `core.tensorboard` or `context.log_metric` must be `cacheable = False`" — would have caught these and #253 without anyone auditing by hand, and would catch the next one.

## Cleared during the same audit, recorded so nobody re-checks

- `PrintNode` — writes to the server console, but its `__log__` is part of the return value and IS replayed on a hit, so the Execution Log entry the user actually reads is preserved.
- `MoELayerNode`, `RewardModelNode` — both call `torch.manual_seed` but save and restore the RNG state around it.
- `KaggleDatasetNode` — mutates `os.environ["KAGGLEHUB_CACHE"]` but restores it in a `finally`, and is `cacheable = False` anyway.
- `VisualizeNode`, `DecisionBoundaryNode`, `ScatterPlot2DNode` — `matplotlib.use("Agg")` is idempotent, figures are closed, `savefig` targets a `BytesIO`.
- `TokenizerNode`, `WordVectorNode`, `EduSentenceEmbeddingNode` — module-level caches, but pure memoization.
- `backend/app/nodes/llm/` — clean apart from `LLMChat`, already `cacheable = False`.

## Acceptance

- For each node listed, either the mutation happens on every run or the node no longer mutates its input.
- Whatever is decided keeps #144's benefit for the read path, or explicitly and visibly gives it up.
- Pinned by a test that runs a graph twice through `execute_graph` and asserts on the mutated object, not on the node's status.

Refs #143, #144, #223, #253.
